Source/DafnyCompilers/AST.dfy

```diff
@@ -1,350 +1,350 @@
-include "CSharpDafnyASTModel.dfy"
+  include "CSharpDafnyASTModel.dfy"
 include "Library.dfy"
 
 module DafnyCompilerCommon.AST {
-    import Lib.Math
-    import Lib.Seq
-    import Microsoft.Boogie
+  import Lib.Math
+  import Lib.Seq
+  import Microsoft.Boogie
+  import C = CSharpDafnyASTModel
+
+  module Types {
     import C = CSharpDafnyASTModel
 
-    module Types {
-      import C = CSharpDafnyASTModel
-
-      type Path = seq<string>
-
-      datatype ClassType = ClassType(className: Path, typeArgs: seq<Type>)
-
-      datatype CollectionKind =
-        | Seq
-        | Set
-        | Multiset
-        | Map(keyType: Type)
-
-      datatype Type =
-        | Unit
-        | Bool
-        | Char
-        | Int
-        | Real
-        | BigOrdinal
-        | BitVector(width: nat)
-        | Collection(finite: bool, kind: CollectionKind, eltType: Type)
-        | Function(args: seq<Type>, ret: Type) // TODO
-        | Class(classType: ClassType)
+    type Path = seq<string>
+
+    datatype ClassType = ClassType(className: Path, typeArgs: seq<Type>)
+
+    datatype CollectionKind =
+      | Seq
+      | Set
+      | Multiset
+      | Map(keyType: Type)
+
+    datatype Type =
+      | Unit
+      | Bool
+      | Char
+      | Int
+      | Real
+      | BigOrdinal
+      | BitVector(width: nat)
+      | Collection(finite: bool, kind: CollectionKind, eltType: Type)
+      | Function(args: seq<Type>, ret: Type) // TODO
+      | Class(classType: ClassType)
+    {
+      // TODO: remove?
+      predicate method NoLeftFunction()
```

Comment:
Expected 2 spaces less indent here

Reply:
and the rest of the predicate

Reply:
I'm not sure I get it. This portion looks correctly indented to me.

Reply:
Agreed. My misreading I guess